core/baseline-core-v1.0-psd01.md

```diff
@@ -2293,12 +2285,18 @@ A workstep MUST have an input, one or more process steps, and an output.
 
 *This is just a well-known convention from business process management frameworks.*
 
+[[R211]](#r211) Testability: A valid workstep is a function that transforms a given input, using one or more transformations of the input into an output. An invalid workstep either does not have an input, or transformations of the input into an output, or an output, or any combination of the three components.
+
 #### **[R212]** 
 The input of a workstep MUST represent a new, proposed state of a state object compliant with the agreement between the agreement counterparties.
 
+[[R212]](#r212) Testability: The input of a workstep where the workstep is represented as a function as in the testability of [[R211]](#r211) can be formatted in a (data) schema of a state object that represents the state of a given agreement between counterparties such as the accumulated amount billed by a Seller to a Buyer. A function input formatted in this way can be validated by the function's logic to be a) conformant to the agreed upon state object schema, and b) to not be equal to the current state object. And the result of these two comparisons can be expressed as a true/false boolean statement as the function output. Therefore, the input can be evaluated to be valid -- a new state object (proposal) -- or not as per the requirement.  
```

Comment:
Please simplify, but noting comment on the requirement itself which might be why this is complex. Partial suggestion depending on how to interpret the overall requirement:

```suggestion
[[R212]](#r212) Testability: The input and output of a workstep can be formatted according to the data schema of a state object that represents the state of a given agreement.

If the input and output are valid representations of a state according to the agreement, and are not equal, they meet the requirement.

If they are equal, or cannot be represented as a valid state object, they do not meet the requirement.
```